src/main/java/com/jeldoclet/XMLNode.java

```diff
@@ -218,28 +217,88 @@ public String toString(String tabs)
+//		out = (gtPat.matcher(out)).replaceAll("&gt;");
+//		out = (aposPat.matcher(out)).replaceAll("&apos;");
+//		out = (quotPat.matcher(out)).replaceAll("&quot;");
+//
+//		return out;
+//	}
+
+	/**
+	 * Returns the string where all non-ascii and <, &, > are encoded as numeric entities. I.e. "&lt;A &amp; B &gt;"
+	 * .... (insert result here). The result is safe to include anywhere in a text field in an XML-string. If there was
+	 * no characters to protect, the original string is returned.
 	 * 
-	 * @param in The input string
-	 * @return The encoded string.
+	 * @param originalUnprotectedString
+	 *            original string which may contain characters either reserved in XML or with different representation
+	 *            in different encodings (like 8859-1 and UFT-8)
+	 * @see https://stackoverflow.com/questions/439298/best-way-to-encode-text-data-for-xml-in-java
+	 * @return
 	 */
-	static protected String encode( String in )
-	{
-		Pattern ampPat = Pattern.compile( "&" );
-		Pattern ltPat = Pattern.compile( "<" );
-		Pattern gtPat = Pattern.compile( ">" );
-		Pattern aposPat = Pattern.compile( "\'" );
-		Pattern quotPat = Pattern.compile( "\"" );
+	 static String encode(String originalUnprotectedString) {
+	    if (originalUnprotectedString == null) {
+	        return null;
+	    }
+	    boolean anyCharactersProtected = false;
 
-		String out = new String( in );
+	    StringBuilder stringBuffer = new StringBuilder(originalUnprotectedString.length());
+	    for (int i = 0; i < originalUnprotectedString.length(); i++) {
+	        char ch = originalUnprotectedString.charAt(i);
 
-		out = (ampPat.matcher(out)).replaceAll("&amp;");
-		out = (ltPat.matcher(out)).replaceAll("&lt;");
-		out = (gtPat.matcher(out)).replaceAll("&gt;");
-		out = (aposPat.matcher(out)).replaceAll("&apos;");
-		out = (quotPat.matcher(out)).replaceAll("&quot;");
+	        if (ch<32 || ch>126) {
+	          // control characters or unicode but not Ascii
+            stringBuffer.append("&#" + (int) ch + ";");
+            anyCharactersProtected = true;
+	        } else
+	           switch (ch)
+            {
+            case '<':
+              stringBuffer.append("&lt;");
+              anyCharactersProtected = true;
+              break;
+            case '>':
+              stringBuffer.append("&gt;");
+              anyCharactersProtected = true;
+              break;
+            case '&':
+              stringBuffer.append("&amp;");
+              anyCharactersProtected = true;
+              break;
+            case '\'':
+              stringBuffer.append("&apos;");
+              anyCharactersProtected = true;
+              break;
+            case '"':
+              stringBuffer.append("&quot;");
+              anyCharactersProtected = true;
+              break;
+            default:
+              stringBuffer.append(ch);
+              anyCharactersProtected = true;
+            }
+	    }
+	    if (anyCharactersProtected == false) {
```

Comment:
`anyCharactersProtected` is always true, as far as I can see. 

I'd just get rid of this `if` block (and the whole `anyCharactersProtected` flag altogether) and always return `stringBuffer.toString()` as this seems like a really minor optimisation to me (if at all).

but if you want to keep it, then, probably, in the default case of the switch it should be:

```
default:
      stringBuffer.append(ch);
      anyCharactersProtected = false;
}
```